ddl/ddl_api.go

```diff
@@ -3588,6 +3584,11 @@ func (d *ddl) dropIndex(ctx sessionctx.Context, ti ast.Ident, isPK bool, indexNa
 	}
 
 	indexInfo := t.Meta().FindIndexByName(indexName.L)
+	var isPK bool
+	if indexName.L == strings.ToLower(mysql.PrimaryKeyName) &&
```

Comment:
Suppose that:
```
create table tt(`primary` int, b int);
alter table tt add index (`primary`);
```
What if 
`alter table tt add primary key(b);`?

Reply:
I think this question and your last question are a problem. I file the issue of #14243. I think we can handle it in the next PR.

Reply:
If we need to follow MySQL name convention, that `primay` is always used by PRIMARY KEY, then we don't need to check `(indexInfo == nil || indexInfo.Primary)`.

Reply:
But this problem may occur with indexes that already exist in the old TiDB cluster. I will add some comments here.